src/components/organisms/FridgeList/FridgeList.jsx

```diff
@@ -0,0 +1,122 @@
+import { Button, List, ListItem, Stack, Typography } from '@mui/material';
+import {
+  CalendarMonthOutlined as CalendarIcon,
+  Instagram as InstagramIcon,
+  LocationOnOutlined as LocationOnOutlinedIcon,
+} from '@mui/icons-material';
+
+import Image from 'next/image';
+import Link from 'next/link';
+import PropTypes from 'prop-types';
+import { getInstagramHandleFromUrl } from 'lib/helperFunctions';
+import typesValidation from 'schema/api/fridge/prop-types';
+import { useMemo } from 'react';
+
+function indexReportsByFridgeId(reports) {
+  const ret = {};
+  for (const report of reports) {
+    ret[report.fridgeId] = report;
+  }
+  return ret;
+}
```

Comment:
Ideally, we should do this in back-end code.
I think we will eventually add pagination to the fridge list, then we will not be able to join the fridge list & reports list on client-side anymore.

Reply:
Interesting! I didn't think of this

Yes we wouldn't be able to combine if we did pagination on the list vs the map.

Reply:
So the entire Fridge database will be stored client side. As of now its a 34kb JSON file. The only thing that changes frequently are the Fridge status reports. This is what the app will request on an ongoing basis. This makes the app usable even when network connectivity is lost. Pagination can be easily handled client side, as there is no partial data request.

Reply:
Sounds good to me. We can do the join on the client side if we have access to the entire data.